### PR TITLE
Fix various reported Save and Export issues

### PR DESCRIPTION
**Pull request by mzieg**
The use-case is to launch ENLIGHTEN, use "Load" to load several previously-saved single-Measurement CSV files, then clicking "Export".

Digging into the code, it looks like at some point ColumnFileParser started storing metadata internally as a list (presumably for use when loading multi-Measurement export files).  This broke single-file load.  Also there seemed to be confusion on field capitalization, and an assumption that metadata would always include wavecal coeffs.

The changes were tested by successfully loading several spectra collected over a span of days and re-exporting them.

### CODE REVIEW

**Comment by mzieg** [comment]
Fixes #162

File: enlighten/parser/ColumnFileParser.py [hunk 1]
```diff
@@ -83,10 +83,20 @@ def parse(self) -> Measurement:
...
         if field not in self.metadata or self.metadata[field] is None:
             return 0
 
-        s = self.metadata[field].strip()
+        v = self.metadata[field]
+        if type(v) is list:
+            if len(v) == 1:
+                s = v[0]
+            else:
+                s = str(v)
```
Comment (samiebee43) [inline enlighten/parser/ColumnFileParser.py:95]: This value `s` will never be safe to convert to float.

File: enlighten/parser/ColumnFileParser.py [hunk 2]
```diff
@@ -83,10 +83,20 @@ def parse(self) -> Measurement:
...
+        else:
+            s = str(v)
+
+        s = s.strip()
```
Comment (samiebee43) [inline enlighten/parser/ColumnFileParser.py:99]: not needed

File: enlighten/parser/ColumnFileParser.py [hunk 3]
```diff
@@ -83,10 +83,20 @@ def parse(self) -> Measurement:
     # ##########################################################################
 
     def get_safe_float(self, field):
```
Comment (samiebee43) [inline enlighten/parser/ColumnFileParser.py:85]: this method should be in Common.py and take the dictionary as an argument

Comment (mzieg) [reply enlighten/parser/ColumnFileParser.py]: I'm hoping we can leave it local, now that it's smaller and simpler (less worthy of re-use)

File: enlighten/parser/ColumnFileParser.py (hunk 2 above)
Comment (mzieg) [reply enlighten/parser/ColumnFileParser.py]: simplified

**Comment by mzieg** [comment]
found some bugs, taking back to draft

**Comment by mzieg** [comment]
Expanded use-case:

- Take 2 spectra A.csv and B.csv
- exit ENLIGHTEN
- DISCONNECT SPECTROMETER
- relaunch ENLIGHTEN
- Load A.csv and B.csv
- Export A and B to make AB.csv
- (visually compare A, B and AB in Excel)
- exit ENLIGHTEN
- relaunch ENLIGHTEN
- load AB.csv
- re-export to create AB2.csv
- exit ENLIGHTEN
- (visually compare AB.csv and AB2.csv in Excel)

**Comment by mzieg** [comment]
Hi Samie, just to forestall some obvious observations:

- ExportFileParser and ColumnFileParser clearly need to share more code, if not be merged into a single package
- Much of enlighten.file_io should probably follow CSVLoader into Wasatch.PY
- We should consider either changing the current CSV (individual and export) file formats, or adding new ones, that can be automatically loaded / generated through Pandas or similar, as long as they remain "intuitively useful" from Excel.  The current file formats were intended to be "easy for humans to use from Excel," but are a little clunky programmatically.  (Of course, that's what the JSON formats were provided for, so programmers wouldn't *need* to futz around with the CSVs.)

All of those are valid long-term goals.  At this point, I'm just trying to fix some clear but staightforward bugs in our current implementation.

File: enlighten/parser/ExportFileParser.py [hunk 4]
```diff
@@ -414,6 +426,10 @@ def process_data(self, values):
                     if len(value) == 0:
                         continue
 
+                    # MZ: honestly not sure how to handle these
+                    if value == "NA":
+                        value = 0
```
Comment (samiebee43) [inline enlighten/parser/ExportFileParser.py:431]: [Suggestion] If it's a float you could use `float('NaN')`. Doesn't work for ints.

**Review by samiebee43 — APPROVED** [review]
I want to avoid using `setattr`, but I'll let it go.

[L199:ExportFileParser](https://github.com/WasatchPhotonics/ENLIGHTEN/blob/97519d5ee9e3bde6569cd7740d152927661300c6/enlighten/parser/ExportFileParser.py#L199)

It certainly makes the code prettier than the 8-line alternative, but it makes regex and static analysis harder.

**Comment by samiebee43** [comment]
All non-resolved contentions, observations, whatever I put into issues which we can deal with later. Feel free to merge and close this thread.

**Pull request merged**